sdk/dotnet/Pulumi/Testing/MockMonitor.cs

```diff
@@ -14,8 +14,9 @@ namespace Pulumi.Testing
     internal class MockMonitor : IMonitor
     {
         private readonly IMocks _mocks;
-        private readonly Serializer _serializer = new Serializer();
-        
+        private readonly Serializer _serializer = new Serializer(excessiveDebugOutput: false);
```

Comment:
The constructor needs to be called on this struct, otherwise the `DependentResources` field was `null` resulting in `NullReferenceException` serializing a dictionary containing a resource.

https://github.com/pulumi/pulumi/blob/d926fe6c1369da41787b3d3cac40c73aa98f4a2b/sdk/dotnet/Pulumi/Serialization/Serializer.cs#L15-L25